File: rtpay_inq/렌탈료조회/PHP/inquiryRentalFee.php
<?php

header("Content-Type: text/html; charset=utf-8"); 


    //step1. 요청을 위한 파라미터 설정

    mt_srand((double)microtime() * 1000000);
    $rand_num = mt_rand(3, 2018);

    $mid = "INIpayTest";
    $timestamp   = date("YmdHis");

    $sid = (string)$mid.(string)$timestamp.(string)$rand_num ;

    $clientType = "MR";
    $amount = "100000";


    //step2. key=value post 전송

    $data = array(
        'mid' => $mid,
        'clientType' => $clientType,
        'sid' => $sid,
        'amount' => $amount
    );

    $post_data = json_encode($data);

 
    $url = "https://inirt.inicis.com/contract/v1/inquiryRentalFee";  
    
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_HTTPHEADER, array('Accept: application/json', 'Content-Type: application/json'));
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
     
    $response = curl_exec($ch);
    curl_close($ch);


//step3. 결과출력

echo $response;

?>

File: rtpay_inq/렌탈료조회/Java/com/inicis/inirt/inquiryRentalFee/inquiryRentalFee.java
package com.inicis.inirt.inquiryRentalFee;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.HashMap;
import java.util.Map;

import org.json.simple.JSONObject;

public class inquiryRentalFee {

	public static void main(String[] args) {

	    //step1. 요청을 위한 파라미터 설정
	    String mid = "INIpayTest"; 
	    String clientType = "MR";
	    
	    SimpleDateFormat sdf=new SimpleDateFormat("yyyyMMddhhmmss");
	    String today = sdf.format(Calendar.getInstance().getTime());
	    
	    double random = Math.random()*1000;
	    String num = Integer.toString((int)random);
	    
	    String sid = "RENTAL"+mid+today+num;
	    String amount = "100000";

		// Request URL
		String apiUrl = "https://inirt.inicis.com/contract/v1/inquiryRentalFee";

	    JSONObject respJson = new JSONObject();
	    respJson.put("mid",mid);
	    respJson.put("clientType",clientType);
	    respJson.put("sid", sid);
	    respJson.put("amount", amount);
		
		//step2. key=value 로 post 요청
		try {
			URL apiurl = new URL(apiUrl);
			HttpURLConnection conn = (HttpURLConnection) apiurl.openConnection();
			
			if (conn != null) {
				conn.setRequestProperty("Content-Type", "application/json; charset=utf-8");
				conn.setRequestMethod("POST");
				conn.setDefaultUseCaches(false);
				conn.setDoOutput(true);
				
				if (conn.getDoOutput()) {
					conn.getOutputStream().write(respJson.toString().getBytes("UTF-8"));
					conn.getOutputStream().flush();
					conn.getOutputStream().close();
				}

				conn.connect();
				
					BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8));
					
					//step3. 요청 결과
					System.out.println(br.readLine());
					br.close();
				}

		}catch(Exception e ) {
			e.printStackTrace();
		} 
	}
}

